src/adagrams.js

```diff
@@ -1,15 +1,139 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letters = [
+  'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
+  'B', 'B',
+  'C', 'C',
+  'D', 'D', 'D', 'D',
+  'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E','E',
+  'F','F',
+  'G', 'G', 'G',
+  'H', 'H',
+  'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 
+  'J',
+  'K',
+  'L', 'L', 'L', 'L',
+  'M', 'M',
+  'N', 'N', 'N', 'N', 'N', 'N',
+  'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O',
+  'P', 'P',
+  'Q',
+  'R', 'R', 'R', 'R', 'R', 'R',
+  'S', 'S', 'S', 'S',
+  'T', 'T', 'T', 'T', 'T', 'T',
+  'U',  'U', 'U', 'U',
+  'V', 'V',
+  'W', 'W',
+  'X',
+  'Y', 'Y',
+  'Z'
+  ];
+
+  
+  const handOfCards = [];
+  
+  for (let i = 0; i < 10; i ++) {
+    let randomLetter = letters[Math.floor(Math.random()*letters.length)];
+    handOfCards.push(randomLetter);
+  
+    let index = letters.indexOf(randomLetter);
+    letters.splice(index, 1);
+  }
+  
+  return handOfCards;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+    // Implement this method for wave 2
+  for (const letter of input) {
+    const upperLetter = letter.toUpperCase();
+
+    if (!(lettersInHand.includes(upperLetter))) {
+      return false;
+    }
+    else {
+      let index = lettersInHand.indexOf(upperLetter);
+      lettersInHand.splice(index, 1);
+    }
+  }return true;
 };
 
+
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+    // Implement this method for wave 3
+  const scoreChart = {
```

Comment:
This can be declared outside of the function so it doesn't have to be recreated each time the function is called.